### DISCUSSION TITLE
[A-L] (19 febbraio 2024) Esercizio 4

### DISCUSSION THREAD

**Discussion by Frisbon** [sapienzastudentsnetwork/progettazione-di-algoritmi#86] [Esami A-L]
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/fc1a28a9-5614-411c-a017-5237ff411e3c)

**Comment by Frisbon** [comment]
Algoritmo della scacchiera. Quello visto dal prof non mi piace perciò uso una versione più semplice $O(n^2)$
![image](https://github.com/sapienzastudentsnetwork/progettazione-di-algoritmi/assets/84731341/ebe85685-225f-40ec-bfd7-13bc3ea14e31)

```py
def scacchiera(C):
    n = len(C);
    T = Matrice DI ZERI (n x n)
    
    #la prima cella deve essere rossa!
    if C[1,1] == "ROSSO": T[1,1] = 1
    else: return 0
    
    #si assume che non esiste indice 0 nella matrice
    for i in range(1,n):
        for j in range(1,n):
           if T[i,j] == "ROSSO":
              if i > 1: #se posso pia da sopra
                  T[i,j] += T[i-1 , j]
              if j > 1: #se posso pia da sinistra
                  T[i,j] += T[i , j-1]
    return T[n,n] 
```

**Comment by CiottoloMaggico** [comment]
![immagine](https://github.com/user-attachments/assets/be5aa4f7-d578-4d8c-bd95-baa7f370eb02)

**Comment by daromacarlo** [comment]
![immagine](https://github.com/user-attachments/assets/9e297998-e532-4844-a3c4-ddd7744adc69)

**Comment by cek753** [comment]
<img width="1536" height="1301" alt="image" src="https://github.com/user-attachments/assets/f9d4efd6-7ae0-4869-93a8-747490962062" />

**Comment by LAMASE** [comment]
Andrebbe chiarito se le mosse possibili sono due come dice o se sono tre visto che fa tre esempi di mosse (orizzontale, verticale e diagonale). Due li chiama "verso destra" e uno "sotto" - sbagliati due su tre, succede anche ai migliori.

Nel caso di tre mosse possibili vanno sommati sia sopra e sinistra, che la diagonale in alto a sin (i-1,j-1) perché è un cammino diverso.
